src/quickfix.c

```diff
@@ -7801,6 +7819,23 @@ set_errorlist(
     return retval;
 }
 
+static int mark_quickfix_user_data(qf_info_T *qi, int copyID) 
+{
+    int abort = FALSE;
+    for (int i = 0; i < LISTCOUNT && !abort; ++i) {
+	qfline_T *qfp;
+	int j;
+        FOR_ALL_QFL_ITEMS(&qi->qf_lists[i], qfp, j) 
```

Comment:
This is functionally correct but inefficient.  The garbage collection runs often.  If a user has multiple quickfix/location lists with a large number of entries (without any user data), then this loop will slow down Vim.  Is it possible to set a flag in the list if any of the item has a user data attached to it?  This way only lists with items that have a user data attached need to be inspected.

Reply:
@yegappan would you say this is a blocker? If so, mind giving a few pointers so this fellow can get this to the finish line?

Reply:
Yes.  This will slow down the garbage collection with a large number of quickfix/location lists/entries. 

To address this, a flag should be added to qf_list_T that indicates whether any of the entries in that list have user data.  By default, this will be set to FALSE.  In the qf_add_entry() function, when copying the user data, set this flag to TRUE.  When running the garbage collection for the quickfix lists, check this flag.  If the flag is set, then do the garbage collection update for the user data in the quickfix items for that list.  If the flag it not set, the list can be skipped.

Reply:
Thanks, I was still not sure where to add the default value, but `qf_new_list` looked reasonable, hope this is correct?